mne/io/base.py

```diff
@@ -2015,12 +2018,15 @@ def _write_raw(fname, raw, info, picks, fmt, data_type, reset_range, start,
         write_int(fid, FIFF.FIFF_REF_FILE_NUM, part_idx - 1)
         end_block(fid, FIFF.FIFFB_REF)
 
-    pos_prev = None
+    pos_prev = fid.tell()
+    if pos_prev > split_size:
+        raise ValueError('file is larger than "split_size" after writing '
+                         'measurement information, you must use a larger '
+                         'value for split size: %s plus enough bytes for '
+                         'the chosen buffer_size' % pos_prev)
+    next_file_buffer = 2 ** 20  # extra cushion for last few post-data tags
     for first in range(start, stop, buffer_size):
         last = first + buffer_size
-        if last >= stop:
-            last = stop + 1
```

Comment:
This should never be inclusive of `stop` so we shouldn't need this. If we do, we have a bug elsewhere I think, as this will make this one buffer bigger than the others which does not make sense.